Automattic/pocket-casts-android
# End of Year: Add tracks

| 📘 Project: #410 |
|:---:|

iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/476

## Description

Adds tracking for EoY feature.

## Testing Instructions

1.  Set `END_OF_YEAR_ENABLED` feature flag to `true` in base.gradle
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. ✅ When the prompt appears ensure `end_of_year_modal_shown` is tracked
4. Tap "View My 2022"
5. ✅ Ensure `end_of_year_stories_shown` is tracked with `source: modal`
6. Tap the upper-right "X" button
7. ✅ Ensure `end_of_year_stories_dismissed ["source": "close_button"]` is tracked
8. Go to Profile and tap the "Your Year in Podcasts" card
9. ✅ Ensure `end_of_year_stories_shown ["source": "profile"]` is tracked
10. ✅  For each story, check that `end_of_year_story_shown` is tracked and that the `story` parameter is correct
11. Open the stories and navigate to the last one
12. Tap "Replay"
13. ✅ Ensure `end_of_year_story_replay_button_tapped` is tracked
14. Tap "Share" on a story
15. ✅ Ensure end_of_year_story_share ["story": "intro"] is tracked
16. Share to Twitter
17. ✅ Check the content and the image for the shared story
18. ✅ Ensure end_of_year_story_shared ["activity": "com.twitter.composer.ComposerActivity", "story": "intro"] is tracked, where activity depends on what action you've done

~~Yet to add~~ 

end_of_year_stories_shown with source: user_login - I'll track this in my next PR which will require login for EoY.  - Added in f5930fe
